Content.Server/Objectives/ObjectivesManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Server.Mobs;
+using Content.Server.Objectives.Interfaces;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Objectives
+{
+    public class ObjectivesManager : IObjectivesManager
+    {
+        [Dependency] private IPrototypeManager _prototypeManager = default!;
+        [Dependency] private IRobustRandom _random = default!;
+
+        public ObjectivePrototype[] GetAllPossibleObjectives(Mind mind)
+        {
+            return _prototypeManager.EnumeratePrototypes<ObjectivePrototype>().Where(objectivePrototype => objectivePrototype.CanBeAssigned(mind)).ToArray();
+        }
+
+        public ObjectivePrototype[] GetRandomObjectives(Mind mind, float maxDifficulty = 3)
```

Comment:
Should you maybe add a maxObjectives arg?

Reply:
am doing that with `maxDifficulty`. that way i can have a more dynamic max that just 3 objectives, as they differ in difficulty